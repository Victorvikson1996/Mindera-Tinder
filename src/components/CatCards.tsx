import React from 'react';
import { View, Text, Image, StyleSheet, Dimensions } from 'react-native';
import { Cat } from '@/types/cattypes';
import { Colors } from '@/utils/Colors';

interface CatCardProps {
  cat: Cat;
  index: number;
}

const { width } = Dimensions.get('window');
const CARD_WIDTH = width * 0.88;
const CARD_RADIUS = 22;
const FOOTER_HEIGHT = 70;

export const CatCard: React.FC<CatCardProps> = ({ cat, index }) => (
  <View style={styles.card}>
    <Image
      source={{ uri: cat.image?.url }}
      style={styles.image}
      resizeMode='cover'
    />
    <View style={styles.footer}>
      <View>
        <Text style={styles.breed}>{cat.name}</Text>
        <Text style={styles.origin}>{cat.origin}</Text>
      </View>
      <Text style={styles.index}>{index}</Text>
    </View>
  </View>
);

const styles = StyleSheet.create({
  card: {
    width: CARD_WIDTH,
    height: CARD_WIDTH + 20,
    borderRadius: CARD_RADIUS,
    overflow: 'hidden',
    backgroundColor: '#fff',
    alignSelf: 'center',
    marginVertical: 12,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 8 },
    shadowOpacity: 0.1,
    shadowRadius: 10,
    elevation: 4
  },
  image: {
    width: '100%',
    height: '100%'
  },
  footer: {
    position: 'absolute',
    bottom: 0,
    left: 12,
    right: 12,
    height: FOOTER_HEIGHT,
    backgroundColor: '#fff',
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,

    paddingHorizontal: 18,
    paddingVertical: 12,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    shadowOffset: { width: 0, height: -1 },
    shadowOpacity: 0.05,
    shadowRadius: 2,
    elevation: 1
  },
  breed: {
    fontSize: 19,
    fontWeight: '700',
    color: Colors.dark,
    marginBottom: 2
  },
  origin: {
    fontSize: 13,
    color: Colors.grey
  },
  index: {
    fontSize: 18,
    color: Colors.dark,
    fontWeight: '700'
  }
});
