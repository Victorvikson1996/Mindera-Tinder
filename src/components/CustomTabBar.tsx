import React from 'react';
import { View, TouchableOpacity, StyleSheet, Dimensions } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { BottomTabBarProps } from '@react-navigation/bottom-tabs';
import { Colors } from '@/utils/Colors';

const { width } = Dimensions.get('window');

const icons = [
  { name: 'paw', route: 'Home', activeColor: Colors.paw },
  {
    name: 'chatbubble-outline',
    route: 'Counter',
    activeColor: Colors.paw
  },
  { name: 'person-outline', route: 'Profile', activeColor: Colors.paw }
];

export const CustomTabBar = ({
  state,
  descriptors,
  navigation
}: BottomTabBarProps) => {
  return (
    <View style={styles.container}>
      <View style={styles.tabBar}>
        {icons.map((icon, idx) => {
          const isFocused = state.index === idx;
          const onPress = () => {
            if (!isFocused) navigation.navigate(icon.route as never);
          };
          return (
            <TouchableOpacity
              key={icon.route}
              accessibilityRole='button'
              accessibilityState={isFocused ? { selected: true } : {}}
              onPress={onPress}
              style={styles.tab}
              activeOpacity={0.8}
            >
              <Ionicons
                name={icon.name as any}
                size={24}
                color={isFocused ? icon.activeColor : Colors.dark}
              />
            </TouchableOpacity>
          );
        })}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    bottom: 32,
    left: 0,
    right: 0,
    alignItems: 'center',
    zIndex: 99
  },
  tabBar: {
    flexDirection: 'row',
    backgroundColor: Colors.white,
    borderRadius: 50,
    width: width * 0.5,
    alignSelf: 'center',
    justifyContent: 'space-around',
    alignItems: 'center',
    paddingVertical: 10,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 4
    },
    shadowOpacity: 0.08,
    shadowRadius: 16,
    elevation: 10
  },
  tab: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 2
  },
  pinkDot: {
    width: 7,
    height: 7,
    borderRadius: 4,
    backgroundColor: Colors.paw,
    marginTop: 4
  }
});
