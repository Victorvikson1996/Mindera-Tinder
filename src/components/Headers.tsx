import React, { useState } from 'react';
import { View, TouchableOpacity, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { Colors } from '@/utils/Colors';

export const CatTinderHeader = () => {
  const [selected, setSelected] = useState<'fire' | 'star'>('fire');

  return (
    <View style={styles.container}>
      <TouchableOpacity
        onPress={() => setSelected('fire')}
        style={[
          styles.iconWrapper,
          selected === 'fire' && styles.activeIconWrapper
        ]}
      >
        <Ionicons
          name='flame'
          size={20}
          color={selected === 'fire' ? Colors.paw : Colors.grey}
        />
      </TouchableOpacity>

      <TouchableOpacity
        onPress={() => setSelected('star')}
        style={[
          styles.iconWrapper,
          selected === 'star' && styles.activeIconWrapper
        ]}
      >
        <Ionicons
          name='star'
          size={20}
          color={selected === 'star' ? Colors.paw : Colors.grey}
        />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignSelf: 'center',
    backgroundColor: '#F2F2F7',
    borderRadius: 32,
    width: 80,
    height: 40,
    padding: 4,
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  iconWrapper: {
    width: 32,
    height: 32,
    borderRadius: 16,
    justifyContent: 'center',
    alignItems: 'center'
  },
  activeIconWrapper: {
    backgroundColor: Colors.white,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.15,
    shadowRadius: 4,
    elevation: 4
  }
});
