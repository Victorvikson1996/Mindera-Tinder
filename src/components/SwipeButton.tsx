import React from 'react';
import { View, TouchableOpacity, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { Colors } from '@/utils/Colors';

interface ButtonProps {
  onLeft: () => void;
  onRight: () => void;
}

export const SwipeButtons: React.FC<ButtonProps> = React.memo(
  ({ onLeft, onRight }) => (
    <View style={styles.row}>
      <TouchableOpacity style={[styles.button, styles.red]} onPress={onLeft}>
        <Ionicons name='close' size={36} color={Colors.paw} />
      </TouchableOpacity>
      <TouchableOpacity style={[styles.button, styles.green]} onPress={onRight}>
        <Ionicons name='heart' size={36} color={Colors.green} />
      </TouchableOpacity>
    </View>
  )
);

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 18
  },
  button: {
    backgroundColor: '#fff',
    borderRadius: 40,
    width: 64,
    height: 64,
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: 18,
    shadowColor: '#aaa',
    shadowOffset: { width: 0, height: 5 },
    shadowOpacity: 0.14,
    shadowRadius: 8,
    elevation: 6
  },
  red: {},
  green: {}
});
