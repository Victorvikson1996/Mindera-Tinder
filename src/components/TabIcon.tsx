import React from 'react';
import { Ionicons } from '@expo/vector-icons';
import { View, StyleSheet } from 'react-native';

interface TabBarIconProps {
  name: React.ComponentProps<typeof Ionicons>['name'];
  focused: boolean;
}

export const TabBarIcon: React.FC<TabBarIconProps> = ({ name, focused }) => {
  const color = focused ? '#e75480' : '#bbb';

  return (
    <View style={styles.iconWrapper}>
      <Ionicons name={name} size={26} color={color} />
      {focused && <View style={styles.dot} />}
    </View>
  );
};

const styles = StyleSheet.create({
  iconWrapper: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  dot: {
    width: 6,
    height: 6,
    borderRadius: 3,
    backgroundColor: '#e75480',
    marginTop: 4
  }
});
