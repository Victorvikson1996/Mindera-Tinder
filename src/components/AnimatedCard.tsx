import React, { useCallback } from 'react';
import { StyleSheet, Dimensions } from 'react-native';
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withSpring,
  runOnJS
} from 'react-native-reanimated';
import { Gesture, GestureDetector } from 'react-native-gesture-handler';
import { CatCard } from './CatCards';
import { Cat } from '@/types/cattypes';

const { width } = Dimensions.get('window');
const SWIPE_THRESHOLD = width * 0.28;

interface Props {
  cat: Cat;
  onSwipeLeft: () => void;
  onSwipeRight: () => void;
}

export const AnimatedCatCard: React.FC<Props> = React.memo(
  ({ cat, onSwipeLeft, onSwipeRight }) => {
    const translateX = useSharedValue(0);
    const startX = useSharedValue(0);

    const panGesture = Gesture.Pan()
      .onStart(() => {
        startX.value = translateX.value;
      })
      .onUpdate((event) => {
        translateX.value = startX.value + event.translationX;
      })
      .onEnd((event) => {
        if (event.translationX > SWIPE_THRESHOLD) {
          translateX.value = withSpring(width, {}, () => {
            runOnJS(onSwipeRight)();
          });
        } else if (event.translationX < -SWIPE_THRESHOLD) {
          translateX.value = withSpring(-width, {}, () => {
            runOnJS(onSwipeLeft)();
          });
        } else {
          translateX.value = withSpring(0);
        }
      });

    const animatedStyle = useAnimatedStyle(() => ({
      transform: [
        { translateX: translateX.value },
        { rotateZ: `${translateX.value / 25}deg` }
      ]
    }));

    return (
      <GestureDetector gesture={panGesture}>
        <Animated.View style={[styles.card, animatedStyle]}>
          <CatCard cat={cat} />
        </Animated.View>
      </GestureDetector>
    );
  }
);

// Add display name for debugging
AnimatedCatCard.displayName = 'AnimatedCatCard';

const styles = StyleSheet.create({
  card: {
    position: 'absolute',
    width: width * 0.88,
    alignSelf: 'center',
    zIndex: 10
  }
});
