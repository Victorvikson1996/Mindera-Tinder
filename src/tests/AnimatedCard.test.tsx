import React from 'react';
import { render } from '@testing-library/react-native';
import { AnimatedCatCard } from '@/components/AnimatedCard'; // Adjust the import path as necessary

describe('AnimatedCard', () => {
  it('renders correctly', () => {
    const mockCat = {
      id: '1',
      name: 'Whiskers',
      origin: 'FR',
      image: { id: 'img1', url: 'cat.png' }
    };
    const { toJSON } = render(
      <AnimatedCatCard
        cat={mockCat}
        onSwipeLeft={() => {}}
        onSwipeRight={() => {}}
        index={0}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
