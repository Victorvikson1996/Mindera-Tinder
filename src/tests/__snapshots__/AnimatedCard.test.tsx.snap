// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedCard renders correctly 1`] = `
<View
  collapsable={false}
  style={
    [
      {
        "alignSelf": "center",
        "marginTop": 4,
        "width": 718,
        "zIndex": 10,
      },
      {
        "transform": [
          {
            "translateX": 0,
          },
          {
            "rotateZ": "0deg",
          },
        ],
      },
    ]
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 22,
        "elevation": 4,
        "height": 680,
        "marginVertical": 12,
        "overflow": "hidden",
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 10,
        "width": 660,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "uri": "cat.png",
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "bottom": 0,
          "elevation": 1,
          "flexDirection": "row",
          "height": 70,
          "justifyContent": "space-between",
          "left": 12,
          "paddingHorizontal": 18,
          "paddingVertical": 12,
          "position": "absolute",
          "right": 12,
          "shadowOffset": {
            "height": -1,
            "width": 0,
          },
          "shadowOpacity": 0.05,
          "shadowRadius": 2,
        }
      }
    >
      <View>
        <Text
          style={
            {
              "color": "#434141",
              "fontSize": 19,
              "fontWeight": "700",
              "marginBottom": 2,
            }
          }
        >
          Whiskers
        </Text>
        <Text
          style={
            {
              "color": "#BFBFC0",
              "fontSize": 13,
            }
          }
        >
          FR
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#434141",
            "fontSize": 18,
            "fontWeight": "700",
          }
        }
      >
        0
      </Text>
    </View>
  </View>
</View>
`;
